### PR TITLE
[4066] Fix image attachments without dimensions not loading

### LINKED ISSUE #4066: Images without dimensions not loading properly
### Context & proposal

It appears that changes in #3176 had an unwanted side effect, making images uploaded without width and height attributes not load properly. Only the images uploaded after the BE feature was implemented have dimensions so this makes older images incompatible.

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Allow all images to load properly regardless of the existence of dimensions.

### PR DESCRIPTION
**Pull request by MarinTolic**
### 🎯 Goal

Closes #4066 

### 🛠 Implementation details

Added and else branch wrapping the image attachment View height in case the attachment does not have fixed dimensions

### 🎨 UI Changes

_Add relevant screenshots_

| Before | After |
| --- | --- |
| fixed_height_before | fixed_height_after |

### 🧪 Testing

<details>

<summary>Patch that logs image dimensions</summary>

```
Index: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/ImageAttachmentsGroupView.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/ImageAttachmentsGroupView.kt b/stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/ImageAttachmentsGroupView.kt
--- a/stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/ImageAttachmentsGroupView.kt	(revision 9c63ff241ad911e34e92240c6fb85f5d11075a58)
+++ b/stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/ImageAttachmentsGroupView.kt	(date 1661163785295)
@@ -19,6 +19,7 @@
 import android.content.Context
 import android.graphics.drawable.Drawable
 import android.util.AttributeSet
+import android.util.Log
 import android.view.View
 import android.view.ViewGroup
 import androidx.constraintlayout.widget.ConstraintLayout
@@ -89,6 +90,7 @@
     }
 
     private fun showOne(first: Attachment) {
+        Log.d("ImageDimens", "originalWidth: ${first.originalWidth} x originalHeight: ${first.originalHeight}")
         removeAllViews()
         val imageAttachmentView = createImageAttachmentView()
         addView(imageAttachmentView)
```

</details>

Pre testing: Apply the patch above and regex Logcat for `ImageDimens`. Make **clean installs** for all test scenarios.

Test Scenario A: Making sure this PR fixes the issue and loads images without `Attachment.originalWidth` and `Attachment.originalHeight`

1. Log in as Amit
2. Go to the channel `Same People`
3. Scroll up until you find images without fixed dimensions (you will see `ImageDimens: originalWidth: null x originalHeight: null` in the log)
4. Confirm that it is loading successfully
5. Compare with `Develop` (should not load successfully there)

Test Scenario B: Making sure that fixed height images load correctly as well

1. Log in as Amit
2. Go to the channel `Fixed image sizes test channel`
3. Scroll faster to load the images. 
4. Verify that the container size prior to loading is the same as the size after loading (no jumping should occur)

Test Scenario C: Making sure that new uploads work well

1. Upload an image to any channel
2. It should load and be previewed normally

#### General
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `main` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

gif

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/ImageAttachmentsGroupView.kt [hunk 1]
```diff
@@ -109,18 +109,13 @@ internal class ImageAttachmentsGroupView : ConstraintLayout {
             if (imageWidth != null && imageHeight != null) {
                 val ratio = (imageWidth / imageHeight).toString()
                 this.setDimensionRatio(imageAttachmentView.id, ratio)
+            } else {
+                constrainHeight(imageAttachmentView.id, LayoutParams.WRAP_CONTENT)
```
Comment (MarinTolic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/ImageAttachmentsGroupView.kt:113]: Useless comment.

Our public bot seems to approve commits to main eagerly so I'm leaving a comment here preventing a merge.

**Review by TGazica — APPROVED** [review]
Works and looks good.

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] UI Changes correct (before & after images)
- [x] Bugs validated (bugfixes)
- [x] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

**Comment by bychkovdmitry** [comment]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [ ] Compose sample runs & works
- [x] UI Changes correct (before & after images)
- [x] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

**Review by bychkovdmitry — APPROVED** [review]
The 3 scenarios from the description work as described

**Pull request merged**